modules/swagger-codegen/src/main/java/com/wordnik/swagger/codegen/examples/ExampleGenerator.java

```diff
@@ -53,12 +65,10 @@ else if(property != null && mediaType.startsWith("application/xml")) {
       }
     }
     else {
-      for(String key: examples.keySet()) {
-        String value = examples.get(key);
-
-        Map<String, String> kv = new HashMap<String, String>();
-        kv.put("contentType", key);
-        kv.put("example", value);
+      for(Map.Entry<String, Object> entry: examples.entrySet()) {
+        final Map<String, String> kv = new HashMap<String, String>();
+        kv.put("contentType", entry.getKey());
+        kv.put("example", Json.pretty(entry.getValue()));
```

Comment:
@tomtit I would recommend you to leave a comment (TODO) in the code so that someone can find out what needs to be done and potentially file a PR for that.